src/tetris/profiler/hardwareController/cameraController.ts

```diff
@@ -0,0 +1,125 @@
+import HardwarePermission from "tetris/profiler/hardwareController/hardwarePermission";
+
+export default class CameraController {
+
+	//region public members
+	public get permissionState(): HardwarePermission {
+		return this._permissionState;
+	}
+	//endregion
+
+	//region public methods
+	public startVideoStream(): Promise<MediaStream> {
+		if (this.permissionState === HardwarePermission.denied) {
+			return Promise.reject(new Error('No Permissions granted'));
+		}
+		return new Promise( (resolve, reject) => {
+			navigator.mediaDevices.getUserMedia({video: true})
+				.then((stream: MediaStream) => {
+					this._permissionState = HardwarePermission.granted;
+					this._videoElement.classList.add('visible');
+					this._videoElement.srcObject = stream;
+					this._videoStream = stream;
+					this._videoElement.play().then(() => resolve(stream));
+				})
+				.catch(reason => {
+					this._permissionState = HardwarePermission.denied;
+					console.log('Can not start video stream. Message: ' + reason);
+					reject(new Error('Can not start video stream'));
+				});
+		});
+	}
+
+	public stopVideoStream(): void {
+		if (!this._videoStream) {
+			return;
+		}
+		this._videoElement.pause();
+		this._videoStream.stop();
+		this._videoElement.src = null;
+	}
+
+	public takeSnapshot(): Promise<string> {
+		if (this._videoStream) {
+			return Promise.resolve(this._takeSnapshot());
+		}
+		return new Promise((resolve, reject) => {
+			this.startVideoStream()
+				.then(() => resolve(this._takeSnapshot()))
+				.catch(error => reject(error));
+		});
+
+	}
+
+	//endregion
+
+	//region constructor
+	private static _instance: CameraController;
```

Comment:
This code is out of place. Please move it below the constructor, to `private members`.

Reply:
it is a static variable and highly related to the constructor (since it implements the singleton pattern). Therefore, we decided to place it in the constructor region